ui/nuxeo-document-comments/nuxeo-document-comment.js

```diff
@@ -0,0 +1,462 @@
+                      ></iron-icon>
+                    </template>
+                  </div>
+                </template>
+                <template is="dom-if" if="[[editing]]">
+                  <div class="input-area">
+                    <paper-textarea
+                      id="inputContainer"
+                      placeholder="[[_computeTextLabel(level, 'writePlaceholder', null, i18n)]]"
+                      value="{{text}}"
+                      max-rows="[[_computeMaxRows()]]"
+                      no-label-float
+                      on-keydown="_checkForEnter"
+                    >
+                    </paper-textarea>
+                    <template is="dom-if" if="[[!_isBlank(text)]]">
+                      <iron-icon
+                        id="submit"
+                        name="submit"
+                        class="main-option opaque"
+                        icon="check"
+                        on-tap="_submitComment"
+                      ></iron-icon>
+                      <nuxeo-tooltip for="submit">[[i18n('comments.submit.tooltip')]]</nuxeo-tooltip>
+                      <iron-icon name="clear" class="main-option opaque" icon="clear" on-tap="_clearInput"></iron-icon>
+                    </template>
+                  </div>
+                </template>
+                <template is="dom-if" if="[[_isSummaryVisible(comment.expanded, comment.numberOfReplies)]]">
+                  <div id="summary" class="horizontal smaller">
+                    <span class="more-content pointer no-selection" on-tap="_expand"
+                      >[[i18n('comments.numberOfReplies', comment.numberOfReplies)]]</span
+                    >
+                    <span class="separator opaque">•</span>
+                    <span class="opaque"
+                      >[[_computeDateLabel(comment, 'lastReplyDate', comment.lastReplyDate, i18n)]]</span
+                    >
+                  </div>
+                </template>
+              </div>
+
+              <template is="dom-if" if="[[comment.expanded]]">
+                <nuxeo-document-comment-thread id="thread" uid="[[comment.id]]" level="[[_computeSubLevel(level)]]">
+                </nuxeo-document-comment-thread>
+              </template>
+            </div>
+          </div>
+        </template>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-comment';
+    }
+
+    static get properties() {
+      return {
+        comment: {
+          type: Object,
+        },
+
+        level: {
+          type: Number,
+          value: 1,
+        },
+
+        truncated: {
+          type: Boolean,
+          computed: '_computeTruncatedFlag(comment.showFull, comment.text,  maxChars)',
+        },
+
+        maxChars: {
+          type: Number,
+          readOnly: true,
+          value: 256,
+        },
+
+        editing: {
+          type: Boolean,
+          readOnly: true,
+          value: false,
```

Comment:
nit: reflecting this to attribute could be useful for styling, although is kind of a _private_ element...

Reply:
Indeed.
Regarding the "private element", I understand the point but I can also foresee the usage of `nuxeo-document-comment' to show a specific comment (kind of read-mode). We may need to perform small changes for that, but seems a valid usage. WDYT?

Reply:
Agree.